src/metabase/models/params/card_values.clj

```diff
@@ -0,0 +1,52 @@
+(ns metabase.models.params.card-values
+  "Code related to getting values for a parameter where its source values is a card."
+  (:require
+    [clojure.string :as str]
+    [metabase.models :refer [Card]]
+    [metabase.query-processor :as qp]
+    [metabase.util.schema :as su]
+    [schema.core :as s]
+    [toucan.db :as db]))
+
+(def ^:dynamic *max-rows*
+  "Maximum number of rows returned when running a card."
+  2000)
```

Comment:
Still would appreciate a comment explaining why 2000 was picked. Even if there's no specific reason, leave that as a comment. Otherwise our future selves will be puzzled by this magic number

Reply:
updated : https://github.com/metabase/metabase/pull/27343/commits/939983670dda1fdd1b01056db144ea6257a246af